OCPBUGS#38352-14: Updated networking matrix table (#83704)
openshift/openshift-docs: modules/install-osp-deploy-dualstack.adoc

// Module included in the following assemblies:
//
// * installing/installing_openstack/installing-openstack-installer-custom.adoc

:_mod-docs-content-type: PROCEDURE
[id="install-osp-deploy-dualstack_{context}"]
= Deploying the dual-stack cluster

For dual-stack networking in {product-title} clusters, you can configure IPv4 and IPv6 address endpoints for cluster nodes. 

.Prerequisites

* You enabled Dynamic Host Configuration Protocol (DHCP) on the subnets.

.Procedure

. Create a network with IPv4 and IPv6 subnets. The available address modes for `ipv6-ra-mode` and `ipv6-address-mode` fields are: `stateful`, `stateless` and `slaac`.
+
[NOTE]
====
The dual-stack network MTU must accommodate both the minimum MTU for IPv6, which is `1280`, and the OVN-Kubernetes encapsulation overhead, which is `100`.
====

. Create the API and Ingress VIPs ports.

. Add the IPv6 subnet to the router to enable router advertisements. If you are using a provider network, you can enable router advertisements by adding the network as an external gateway, which also enables external connectivity.

> . For an IPv4/IPv6 dual-stack cluster where you set IPv4 as the primary endpoint for your cluster nodes, edit the `install-config.yaml` file in a similar way to the following example:

Reviewer: ```suggestion
. For an IPv4/IPv6 dual-stack cluster where you set IPv4 as the primary endpoint for your cluster nodes, edit the `install-config.yaml` file like the following example:
```
You could also use "as shown in" or something similar.

        name: dualstack
        id: network-id
----
<1> Dual-stack clusters are supported only with the `TechPreviewNoUpgrade` value.
<2> You must specify an IP address range in the `cidr` field for both IPv4 and IPv6 address families.
<3> Specify the virtual IP (VIP) address endpoints for the Ingress VIP services to provide an interface to the cluster.
<4> Specify the virtual IP (VIP) address endpoints for the API VIP services to provide an interface to the cluster.
<5> Specify the dual-stack network details that are used by all the nodes across the cluster.
<6> The Classless Inter-Domain Routing (CIDR) of any subnet specified in this field must match the CIDRs listed on `networks.machineNetwork`.

> <7> You can specify a value for either `name` or `id`, or both.

Reviewer: ```suggestion
<7> You can specify a value for `name`, `id`, or both.
```
